geps/gep-1619.md

```diff
@@ -159,47 +264,398 @@ Generally, the implementation API programs the dataplane API; however these two
 
-## API
-
-### Session Persistence API
-
-The majority of implementations and dataplanes support configuring cookies for session persistence while support for header-based session persistence is limited.
+### Session Affinity in K8S Services
 
-To organize potential Gateway API configuration of cookie functionality, the next section will break down configuration into specific categories:
-- Session Cookie Configuration
-- Session Cookie Strategy
+Kubernetes provides an API that allows you to enable [session affinity](https://kubernetes.io/docs/reference/networking/virtual-ips/#session-affinity)
+on service objects. It ensures consistent sessions by utilizing the client's IP address and also offers the option to
+set a timeout for the maximum session duration. Implementations of Gateway API don't rely on the service IP, so this
+particular application of session affinity doesn't impact the design of the GEP. However, we recommend that users enable
+session affinity using the Kubernetes service object to for applications requiring persistence, particularly due to the
+fact that in-cluster traffic will make use of the service IP.
 
-**Session Cookie Configuration**
-
-Session cookie configuration are the parameters and attributes the dataplane uses to create a cookie for establishing a session.
-
-A set-cookie request can be broken down into [3 main components](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie):
-- Cookie Name
-- Cookie Value
-- Cookie Attributes
+## API
 
-Only the Cookie Name and Cookie Value are required. All Cookie Attributes are optional. Attributes are simply key=value pairs with the value as optional for some attributes.
+### Session Persistence API
 
-The set-cookies attributes defined by [rfc6265](https://www.rfc-editor.org/rfc/rfc6265#section-5.2) are:
+In this section, we will explore the questions and design elements associated with a session persistence API.
+
+#### GO
+
+```go
+// SessionPersistencePolicy provides a way to define session persistence rules
+// for a service or route.
+//
+// Support: Core
+type SessionPersistencePolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of SessionPersistencePolicy.
+    Spec SessionPersistencePolicySpec `json:"spec"`
+
+    // Status defines the current state of SessionPersistencePolicy.
+    Status SessionPersistencePolicyStatus `json:"status,omitempty"`
+}
+
+// SessionPersistencePolicySpec defines the desired state of
+// SessionPersistencePolicy.
+// Note: there is no Override or Default policy configuration.
+type SessionPersistencePolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    // HttpRoutes, GRPCRoute, Services are the only valid API target
```

Comment:
I think it is `HTTPRoute` and not `HttpRoute` (https://gateway-api.sigs.k8s.io/api-types/httproute/) . Needs to be changed everywhere.

Reply:
Ah yea good catch thanks. Done.